lib/utils.js

```diff
@@ -423,6 +425,19 @@ function mergeSettings(target, src) {
 }
 
 /**
+ * TODO jsdoc
+ */
+function rankArrayElements(array, rank) {
+  if (!Array.isArray(array) || !Number.isFinite(rank))
+    return array;
+
+  array.forEach(function(el) {
+    el.rank = rank;
+  });
+  return array;
+}
+
+/**
```

Comment:
new helper to add `rank` property to all array elements. Needs to be tinkered to check if elements are objects